### PR TITLE
monitor status of slot swap operation

### PR DESCRIPTION
**Pull request by dikhakha**

### CODE REVIEW

File: Tasks/Common/azurerest-common/azurerestutility.ts [hunk 1]
```diff
@@ -278,6 +278,49 @@ export async function updateWebAppAppSettings(SPN, webAppName: string, resourceG
...
+    var deferred = Q.defer();
+    var attempts = 0;
+    var poll = async function() {
+        if (attempts < 20) {
+            attempts++;
+            await  getOperationStatus(SPN, webAppName, resourceGroupName, sourceSlot, url).then((status) => {
+                if (status === 200) {
+                    deferred.resolve(tl.loc("Successfullyswappedslots"));
+                }
+                else {
```
Comment (sachinma) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: The polling should only continue if the status code is 202.

File: Tasks/AzureAppServiceManage/task.json [hunk 2]
```diff
@@ -187,6 +187,8 @@
         "CouldnotfetchacccesstokenforAzureStatusCode": "Could not fetch acccess token for Azure App Service. StatusCode %s (%s)",
         "WebAppDoesntExist": "Webapp '%s' doesn't exist.",
         "UnabletoretrieveWebAppID": "Unable to retrieve connection details for Azure App Service:'%s'. Status Code: %s",
-        "StartingSwapSlot": "Requesting swap slot for Web App : %s"
+        "StartingSwapSlot": "Requesting swap slot for Web App : %s",
+        "SwaSlotpOperationInProgress": "Swap operation for Web App %s is in progress",
```
Comment (sachinma) [inline Tasks/AzureAppServiceManage/task.json]: Typo

File: Tasks/Common/azurerest-common/azurerestutility.ts (hunk 1 above)
Comment (dikhakha) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: done

File: Tasks/Common/azurerest-common/azurerestutility.ts [hunk 3]
```diff
@@ -278,6 +278,52 @@ export async function updateWebAppAppSettings(SPN, webAppName: string, resourceG
...
+
+function monitorSlotSwap(SPN, webAppName, resourceGroupName, sourceSlot, targetSlot, url) {
+    var deferred = Q.defer();
+    var attempts = 0;
+    var poll = async function() {
+        if (attempts < 20) {
+            attempts++;
+            await  getOperationStatus(SPN, webAppName, resourceGroupName, sourceSlot, url).then((status) => {
+                if (status === 200) {
+                    deferred.resolve(tl.loc("Successfullyswappedslots"));
```
Comment (sachinma) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: do you want to resolve with true for success

File: Tasks/Common/azurerest-common/azurerestutility.ts [hunk 4]
```diff
@@ -298,16 +344,21 @@ export async function swapWebAppSlot(SPN, resourceGroupName: string, webAppName:
...
             deferred.reject(error);
         }
-        if(response.statusCode === 202) {
-            deferred.resolve(tl.loc("Successfullyswappedslots", webAppName, sourceSlot, targetSlot));
+        else if(response.statusCode === 202) {
+            await monitorSlotSwap(SPN, webAppName, resourceGroupName, sourceSlot, targetSlot, response.headers.location).then(() => {
+                deferred.resolve(tl.loc("Successfullyswappedslots", webAppName, sourceSlot, targetSlot));
+            }).catch((error) => {
+                tl.warning(error);
+                deferred.resolve(tl.loc("Successfullyswappedslots", webAppName, sourceSlot, targetSlot));
```
Comment (sachinma) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: Why should it be successful message

Comment (sachinma) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: Why should it be successful message

File: Tasks/Common/azurerest-common/azurerestutility.ts (hunk 3 above)
Comment (dikhakha) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: will do that.

File: Tasks/Common/azurerest-common/azurerestutility.ts (hunk 4 above)
Comment (dikhakha) [inline Tasks/Common/azurerest-common/azurerestutility.ts]: done. if operation status is not completed, task will end successfully with warning message.

File: Tasks/AzureAppServiceManage/Strings/resources.resjson/en-US/resources.resjson [hunk 5]
```diff
@@ -44,5 +44,6 @@
   "loc.messages.CouldnotfetchacccesstokenforAzureStatusCode": "Could not fetch acccess token for Azure App Service. StatusCode %s (%s)",
   "loc.messages.WebAppDoesntExist": "Webapp '%s' doesn't exist.",
   "loc.messages.UnabletoretrieveWebAppID": "Unable to retrieve connection details for Azure App Service:'%s'. Status Code: %s",
-  "loc.messages.StartingSwapSlot": "Requesting swap slot for Web App : %s"
+  "loc.messages.StartingSwapSlot": "Requesting swap slot for Web App : %s",
+  "loc.messages.UnableToMonitorSlotSwapStatus": "Unable to monitor slot swap status for Web App: %s. Status Code: %s"
```
Comment (sachinma) [inline Tasks/AzureAppServiceManage/Strings/resources.resjson/en-US/resources.resjson]: Can you improve the message a little.

File: Tasks/AzureAppServiceManage/azureappservicemanage.ts [hunk 6]
```diff
@@ -5,13 +5,16 @@ var azureRmUtil = require('azurerest-common/azurerestutility.js');
...
+    try {
+        var isSlotSwapSuccess = await azureRmUtil.swapWebAppSlot(endPoint, resourceGroupName, webAppName, sourceSlot, targetSlot, preserveVnet);
+        if (isSlotSwapSuccess) {
+            tl._writeLine(tl.loc("Successfullyswappedslots", webAppName, sourceSlot, targetSlot));
+        }
     }
-    if(sourceSlot === targetSlot){
-        throw new Error(tl.loc("SourceAndTargetSlotCannotBeSame"));
+    catch (errorMessage) {
+        tl.error(errorMessage);
```
Comment (sachinma) [inline Tasks/AzureAppServiceManage/azureappservicemanage.ts]: Do you need to do tl.error if you are throwing the error again. In that case you wont need the try-catch block in this function

**Pull request merged**